### ISSUE TITLE
[Enhancement]  Strict and warn checks

### ISSUE THREAD

**Issue by TheFlexican** [aws-cloudformation/cloudformation-guard#194]
**Is your feature request related to a problem? Please describe.**

In version 1.0, we could use warn and strict checks for our rules. We would have separate rule files for both. When our pipeline ran, it would display the warnings in de build log and email the warnings to the dev., the pipeline would continue. With strict checks, it would do the same except it would fail the pipeline and stop the build process.

**Describe the solution you'd like**

Possibility to specify on a rule level if the rule is either strict or a warn check.

**Describe alternatives you've considered**

Having separate rule directories for warn and strict checks, and do something with exit codes.
It could work, but not the cleanest solution in my opinion. 

**Additional context**

Would love to hear your thoughts on this, and if you maybe have an alternative solution.

Thank you!

**Comment by dchakrav-github** [comment]
@TheFlexican `strict` mode in 1.0 was more behaviorally inconsistent. As an example 

```
AWS::EC2::Volume Encrypted == true
```
When we test it with the following payload 

```
Resources:
   vol1;
      Type: AWS::EC2::Volume
      Properties:
         Encrypted: true
   vol2;
      Type: AWS::EC2::Volume               
```
Should vol2 fail as Encrypted was absent? Currently, without `--strict-checks` in 1.0 it will PASS and with it will FAIL. It does not warn the user that this is even an issue in the first place to fix. Behavioral inconsistencies based on flags sent to the tool can lead to unintended outcomes, and is sometimes a flag away from an incorrect outcome or a security posture being incorrect. 

In 2.0 everything is consistent and will fail for vol2. If I understand the need, we have different sets of rules, some are best practices with soft enforcement and some are hard requirements. They can belong together. 
1. One option is to annotate the rule as you suggested. That would work, and the tool's exit code would reflect that. 
2. The second option is to provide the tool with another input that qualifies soft vs hard failures by rule name.

(2) provides the flexibility to determine on a per pipeline/application basis what is considered best practice/soft/warning vs what is a hard failure. As a rule author, it is difficult to determine whether its intent is soft or hard fail. It is often dependent on the assembly. We were thinking in this direction. Thoughts?

**Comment by TheFlexican** [comment]
What we did now, we created separate directories for critical and best practice rules. The output of both commands are in json and parsed by a self written Lambda. This lambda will a. Print out the error like version cfn-guard version 1 did b. based on the the type of rules it was scanning either exit with an error making codebuild stop, or exist nicely and let codebuild continue. For both actions the Lambda will use AWS SES with specific templates to inform the user who did the commit there were either errors or warnings with their template.
